src/amber/controller/helpers/responders.cr

```diff
@@ -13,30 +13,35 @@ module Amber::Controller::Helpers
       ACCEPT_SEPARATOR_REGEX = /,|,\s/
 
       @requested_responses : Array(String)
-      @available_responses = Hash(String, String).new
+      @available_responses = Hash(String, String | Proc(String) | Proc(Int32)).new
       @type : String? = nil
 
       def initialize(@requested_responses)
       end
 
       # TODO: add JS type similar to rails.
-      def html(html : String)
+      def html(html : String | Proc(String) | Proc(Int32))
```

Comment:
what's the relevance of a proc returning an Int32?

Reply:
If your proc redirects the last statement in the redirect code path is setting the response status code (an Int32) so that is the Proc return type

Reply:
Replace `Proc(String) | Proc(Int32)` with `ProcType` in this file